### PR TITLE
BRO: The Temporal Anchor and support

### PR DESCRIPTION
**Pull request by Northmoc**
The AI for this could be pretty wild... but honestly even just with regular Scry logic it should generate some interesting advantage that the AI could benefit from.

### CODE REVIEW

**Review by Agetian — APPROVED** [review]

**Comment by Hanmac** [comment]
Wouldn't it be easier to use ToBottom as CardCollection?

Or what happens if you shuffle the library in response to the trigger?

File: forge-gui/res/cardsfolder/upcoming/the_temporal_anchor.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Name:The Temporal Anchor
+ManaCost:3 U U U
+Types:Legendary Artifact
+T:Mode$ Phase | Phase$ Upkeep | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigScry | TriggerDescription$ At the beginning of your upkeep, scry 2.
+SVar:TrigScry:DB$ Scry | ScryNum$ 2
+T:Mode$ Scry | ValidPlayer$ You | ToBottom$ True | Execute$ TrigExile | TriggerDescription$ Whenever you choose to put one or more cards on the bottom of your library while scrying, exile that many cards from the bottom of your library.
+SVar:TrigExile:DB$ Dig | DigNum$ X | ChangeNum$ All | FromBottom$ True | DestinationZone$ Exile
+SVar:X:TriggerCount$ScryBottom
+S:Mode$ Continuous | Condition$ PlayerTurn | MayPlay$ True | Affected$ Card.ExiledWithSource | AffectedZone$ Exile | Description$ During your turn, you may play cards exiled with CARDNAME.
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/the_temporal_anchor.txt:9]: The problem is unless ExiledWithSource would be fixed with a timestamp check you can flicker the Anchor and still play the old cards

Comment (Hanmac) [reply forge-gui/res/cardsfolder/upcoming/the_temporal_anchor.txt:9]: For now, you need to remember the exiled cards

**Comment by Northmoc** [comment]
> Wouldn't it be easier to use ToBottom as CardCollection?
> 
> Or what happens if you shuffle the library in response to the trigger?
> 
> Ups the card explicit said number

I did mess with it as a CardCollection for a minute... we can refactor as needed depending on future card designs.

File: forge-gui/res/cardsfolder/upcoming/the_temporal_anchor.txt (hunk 1 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/upcoming/the_temporal_anchor.txt]: Good call – thanks, you two

**Comment by Hanmac** [comment]
> > Wouldn't it be easier to use ToBottom as CardCollection?
> > Or what happens if you shuffle the library in response to the trigger?
> > Ups the card explicit said number
> 
> I did mess with it as a CardCollection for a minute... we can refactor as needed depending on future card designs.

no need! the Card explicit doesn't care about the Cards that got scryed to bottom, but just wants to exile the same number

**Review by tool4ever — APPROVED** [review]

**Pull request merged**